updated path to use {logoPath} instead of string

sohamtembhurne opened the pull request:
Fixes #3965 
Updated path to use {logoPath} from `../../../sites/shared/components/logos/freesewing.mjs` instead of lengthy string. But there are other instances of the lengthy strings as well. Let me know if I should update them as well.

joostdecock commented on plugins/plugin-annotations/src/logo.mjs:
```diff
@@ -1,8 +1,10 @@
     name: 'logo',
     def: (scale) =>
-      `<g id="logo" transform="scale(${scale}) translate(-23 -36)"><path class="logo" stroke="none" fill="currentColor" d="M 35.222,0 C 34.233,0.703 34.284,0.613 33.485,0.874 31.653,1.473 29.896,1.144 27.811,0.97 27.184,0.9 26.562,0.859 25.955,0.855 22.89,0.834 20.287,1.733 19.794,4.243 18.885,4.794 18.049,5.461 17.221,6.129 15.453,7.524 14.122,9.229 13.214,11.284 11.974,14.319 13.094,17.576 13.6 ...
+      `<g id="logo" transform="scale(${scale}) translate(-23 -36)"><path class="logo" stroke="none" fill="currentColor" d=${logoPath} /></g>`,
```
Seems like you forgot to quote the path string

sohamtembhurne replied:
If you look closely, this is using template literals - `<g> ...        ${logoPath}         ...</g>`, so it doesn't need quotes as logoPath is itself a string in `../../../sites/shared/components/logos/freesewing.mjs`

joostdecock replied:
Yes, but this will inject the string without quotes. So your code will generate this:

```svg
<path class="logo" stroke="none" fill="currentColor" d=M 35.222,0 C 34.233,0... />
```

But what we need is valid SVG output, like this:
```
<path class="logo" stroke="none" fill="currentColor" d="M 35.222,0 C 34.233,0..." />
```

So, while it is true that from a pure Javascript point of view, it is not required to put quotes here, it would result in invalid XML/SVG syntax. That's why that quotes are required.

sohamtembhurne replied:
I have made the required changes.

joostdecock reviewed (changes requested):
If there's other locations, then sure.
We need to check the viewBox though. Because the string you replaced is not the same as this one. So we may need to apply a transform.

sohamtembhurne commented:
> If there's other locations, then sure. We need to check the viewBox though. Because the string you replaced is not the same as this one. So we may need to apply a transform.

I analyzed these svgs. There are two different logo strings being used. First the one that I replaced in `logo.mjs` and another the one present in `../../../sites/shared/components/logos/freesewing.mjs`. There are two ways we can encounter this - 
1. To transform the path `logoPath` that I imported from `freesewing.mjs`. This can be done by scaling it 2 times and translating -2px left. 
2. We can create another string in `freesewing.mjs`, consisting of the path that I replaced. Since this path is used in around 8 files. 

In my opinion, it will be better to create a new path (new path refers to the one I replaced) variable.

joostdecock commented:
> > If there's other locations, then sure. We need to check the viewBox though. Because the string you replaced is not the same as this one. So we may need to apply a transform.
> 
> I analyzed these svgs. There are two different logo strings being used. First the one that I replaced in `logo.mjs` and another the one present in `../../../sites/shared/components/logos/freesewing.mjs`. There are two ways we can encounter this -
> 
>     1. To transform the path `logoPath` that I imported from `freesewing.mjs`. This can be done by scaling it 2 times and translating -2px left.
> 
>     2. We can create another string in `freesewing.mjs`, consisting of the path that I replaced. Since this path is used in around 8 files.
> 
> 
> In my opinion, it will be better to create a new path (new path refers to the one I replaced) variable.

The goal is to have 1 single path string for the logo. And then apply transforms based on the use case.
If you'd like to tackle that, then go ahead.

sohamtembhurne commented:
> > > If there's other locations, then sure. We need to check the viewBox though. Because the string you replaced is not the same as this one. So we may need to apply a transform.
> > 
> > 
> > I analyzed these svgs. There are two different logo strings being used. First the one that I replaced in `logo.mjs` and another the one present in `../../../sites/shared/components/logos/freesewing.mjs`. There are two ways we can encounter this -
> > ```
> > 1. To transform the path `logoPath` that I imported from `freesewing.mjs`. This can be done by scaling it 2 times and translating -2px left.
> > 
> > 2. We can create another string in `freesewing.mjs`, consisting of the path that I replaced. Since this path is used in around 8 files.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > In my opinion, it will be better to create a new path (new path refers to the one I replaced) variable.
> 
> The goal is to have 1 single path string for the logo. And then apply transforms based on the use case. If you'd like to tackle that, then go ahead.

I have transformed the logo based on the use case.

joostdecock commented:
@all-contributors please add @sohamtembhurne for code
